### ISSUE TITLE
[webstorm] TypeError [ERR_INVALID_ARG_TYPE]: The "path" argument must be of type string. Received type boolean

### ISSUE THREAD

**Issue by devmondo** [wallabyjs/quokka#367]
### Issue description or question
I just installed quokka in webstorm and running even an empty file i get the error below.

i have purchased a license and activated it.

i uninstalled/ reinstalled, restarted my machine to no avail.

### Quokka.js Console Output

```
TypeError [ERR_INVALID_ARG_TYPE]: The "path" argument must be of type string. Received type boolean
    at assertPath (path.js:39:11)
    at Object.join (path.js:432:7)
    at WebSocketServer.emit (events.js:182:13)
    at Server.emit (events.js:182:13)
    at onParserExecuteCommon (_http_server.js:552:14)
    at onParserExecute (_http_server.js:500:3)

```

### Code editor version

webstorm 2019.1
### OS name and version

Windows 10

**Comment by ArtemGovorov** [comment]
Please try updating to the latest Quokka plugin version 1.0.162, and running the tool again. The error message should contain more details now, please share the new message.

**Comment by devmondo** [comment]
@ArtemGovorov thank you very much now the error is clear, but how to fix it :) 

```bash
Error: Failed to initialize Quokka paths (homeDir: C:\Users\me, cwd: C:\projects\test): TypeError [ERR_INVALID_ARG_TYPE]: The "path" argument must be of type string. Received type boolean
    at assertPath (path.js:39:11)
    at Object.join (path.js:432:7)
    at WebSocketServer.emit (events.js:182:13)
    at Server.emit (events.js:182:13)
    at onParserExecuteCommon (_http_server.js:552:14)
    at onParserExecute (_http_server.js:500:3)
    at WebSocketServer.emit (events.js:182:13)
    at Server.emit (events.js:182:13)
    at onParserExecuteCommon (_http_server.js:552:14)
    at onParserExecute (_http_server.js:500:3)
```

**Comment by ArtemGovorov** [comment]
@devmondo Thanks for checking. Unfortunately we still can't see what's wrong, so have added some more logging. Please update to the latest plugin version 1.0.163 and share the latest error message.

**Comment by devmondo** [comment]
i updated and the same error
```bash
Error: Failed to initialize Quokka paths (homeDir: C:\Users\me, cwd: C:\projects\test): TypeError [ERR_INVALID_ARG_TYPE]: The "path" argument must be of type string. Received type boolean
    at assertPath (path.js:39:11)
    at Object.join (path.js:432:7)
    at WebSocketServer.emit (events.js:182:13)
    at Server.emit (events.js:182:13)
    at onParserExecuteCommon (_http_server.js:552:14)
    at onParserExecute (_http_server.js:500:3)
    at WebSocketServer.emit (events.js:182:13)
    at Server.emit (events.js:182:13)
    at onParserExecuteCommon (_http_server.js:552:14)
    at onParserExecute (_http_server.js:500:3)

```

**Comment by ArtemGovorov** [comment]
Sorry, it looks like the change hasn't made it to the latest version. Please update your Quokka plugin version 1.0.163, and the error message should change.

**Comment by devmondo** [comment]
i updated again here is the error
```bash
Error: Failed to initialize Quokka paths
    homeDir: C:\Users\me, 
    cwd: C:\projects\test
    globalSettingsDir: C:\Users\me\.quokka
    localSettingsFile: C:\projects\test\.quokka
    quokkaFileName: true
    init status: true, true, undefined
TypeError [ERR_INVALID_ARG_TYPE]: The "path" argument must be of type string. Received type boolean
    at assertPath (path.js:39:11)
    at Object.join (path.js:432:7)
    at WebSocketServer.emit (events.js:182:13)
    at Server.emit (events.js:182:13)
    at onParserExecuteCommon (_http_server.js:552:14)
    at onParserExecute (_http_server.js:500:3)
    at WebSocketServer.emit (events.js:182:13)
    at Server.emit (events.js:182:13)
    at onParserExecuteCommon (_http_server.js:552:14)
    at onParserExecute (_http_server.js:500:3)
```

after looking through the error more, I found the problem!
in error above: localSettingsFile: `C:\projects\test\.quokka`
which is correct, but what I am doing is that I am using webstorm `add content root` feature in settings to be able to open 2 projects in one window.
so I have webstorm main project as `C:\projects\test\`  and I added content root directory `C:\projects\test-playground`. so when i run quokka on say `C:\projects\test-playground\play.js`, quokka is resolving to `C:\projects\test\.quokka`  and not `C:\projects\test-playground\.quokka`. that is the problem! 

so to solve this, quokka needs to resolve the real project path when you are using ` dd root content root feature.  

hope this helps to find a fix :)

**Comment by smcenlly** [comment]
Thanks for providing the additional detail. We were able to work out what was broken. When loading non-scratch files, we had some logic to convert the file path to be relative to the project root. In your case what was happening we were stripping the start of your path, `C:\projects\test`, which ended up sending `-playground\filename.js` to quokka runtime, messing up the input arguments as we only expected project files to be loaded.

This has now been fixed and is available in Quokka for IntelliJ v1.0.166.

*Please note:* when running in files outside of your project root (as determined by IntelliJ), the `.quokka` settings file and any referenced node modules will be loaded from the project root, not `C:\projects\test-playground`.
